src/libraries/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/AppleCCCryptor.cs

```diff
@@ -174,21 +184,35 @@ private unsafe int CipherUpdate(byte[] input, int inputOffset, int count, byte[]
             ProcessInteropError(ret, ccStatus);
         }
 
-        private Interop.AppleCrypto.PAL_ChainingMode GetPalChainMode(CipherMode cipherMode)
+        private Interop.AppleCrypto.PAL_ChainingMode GetPalChainMode(Interop.AppleCrypto.PAL_SymmetricAlgorithm algorithm, CipherMode cipherMode, int feedbackSizeInBytes)
         {
-            switch (cipherMode)
+            switch ((algorithm, cipherMode, feedbackSizeInBytes))
             {
-                case CipherMode.CBC:
+                case (_, CipherMode.CBC, _):
                     return Interop.AppleCrypto.PAL_ChainingMode.CBC;
-                case CipherMode.ECB:
+                case (_, CipherMode.ECB, _):
                     return Interop.AppleCrypto.PAL_ChainingMode.ECB;
+                    // let's be consistent across all platforms,
+                    // and do not support CFB for anything else than AES.
+                case (_, CipherMode.CFB, 1):
+                    return Interop.AppleCrypto.PAL_ChainingMode.CFB8;
+                case (Interop.AppleCrypto.PAL_SymmetricAlgorithm.AES, CipherMode.CFB, 16):
+                case (Interop.AppleCrypto.PAL_SymmetricAlgorithm.TripleDES, CipherMode.CFB, 8):
+                    return Interop.AppleCrypto.PAL_ChainingMode.CFB;
+                case (_, CipherMode.CFB, _):
+                    throw new PlatformNotSupportedException(SR.Format(SR.Cryptography_CipherModeFeedbackNotSupported, feedbackSizeInBytes * 8, cipherMode));
```

Comment:
Okay, my fault here. 

The `CryptographicException()` is thrown directly at `CipherMode` setter. However, that would not work here, since the success/failure is determined based on multiple properties. 

Before my changes, this exception could never happen, and can be checked just with Debug.Fail. I need to throw `CryptographicException` here now.